packages/mui-material/src/utils/getChildRef.js

```diff
@@ -0,0 +1,10 @@
+import * as React from 'react';
+
+export default function getChildRef(children) {
+  if (!children || !React.isValidElement(children) || React.Children.count(children) !== 1) {
```

Comment:
This feels wrong to me. The prop type is already here to validate this constraint, but without bundle size or untiled overhead in production.

```suggestion
```

Reply:
fair, since components using `getChildRef` handling this, we might not need checking count here. removed -> https://github.com/mui/material-ui/pull/42818/commits/916818e4fd47e7fd19924395a3ee6053d3d64402

Reply:
> fair, since components using getChildRef handling this, we might not need checking count here. 

@sai6855 It's not what I had in mind though.

If `React.isValidElement(children)` is false, then we return null.
If `React.isValidElement(children)` is true, then we check `React.Children.count(children)` but it will always return 1  since it's children is a valid element.
So `|| React.Children.count(children) !== 1` always wastes CPU/network work, so it should be removed.

Now, on the use of `|| !React.isValidElement()`, I think that it really depends on the DX we want to provide.

If we want to keep the same DX tradeoff as before, we should remove it, it slows the runtime with nothing in exchange. 
If we want to make it so that even if developers can render a component without a children, get a console.error, type error not no exception that developers could be deceived by, then we could either decide to keep it, or probably better, to add a `if (process.env.NODE_ENV !== 'production' && React.isValidElement(children)) { return null }` in the render body of the function. In which case we should also remove it.

So IMHO

```suggestion
```

would be ideal.

Reply:
thanks for detailed explaintation, opened PR to refactor -> https://github.com/mui/material-ui/pull/43022

TL,DR; of PR is

```diff
-  if (!child || !React.isValidElement(child)) {
-    return null;

+  if (process.env.NODE_ENV !== 'production') {
+   if (!React.isValidElement(child)) {
+      console.error(
+        [
+          "MUI: getChildRef doesn't accept array as a child.",
+          'Consider providing a single element instead.',
+        ].join('\n'),
+      );
+      return null;
+    }
```